internal/common/voting/election.go

```diff
@@ -86,8 +86,114 @@ func (e *Election) CloseBallot() shared.ClientID {
 }
 
 func (e *Election) bordaCountResult() shared.ClientID {
-	// TODO implement Borda count winner selection method.
-	return e.pluralityResult()
+	// Implement Borda count winner selection method
+	//(may need to modify if methods design is changed)
+	var candidatesNumber int = 0
+	var islandsNumber int
+	var votesLayoutElect map[int][]int
+	votesSliceSquare := e.votes
+	candidatesNumber = len(e.islandsToVote)
+	islandsNumber = len(votesSliceSquare)
+
+	//Initialize votesLayoutMap
+	for i := 1; i < islandsNumber+1; i++ {
+		for j := 0; j < candidatesNumber; j++ {
+			votesLayoutElect[i] = append(votesLayoutElect[i], 0)
+		}
+	}
+	//Transfer e.votes to votesLayoutMap with type "int"
+	for i := 0; i < islandsNumber; i++ {
+		scoreInit := candidatesNumber + 1
+		for j := 0; j < candidatesNumber; j++ {
+			for k := 0; k < candidatesNumber; k++ {
+				if votesSliceSquare[i][j] == e.islandsToVote[k] {
+					votesLayoutElect[i+1][k] = scoreInit
+					scoreInit -= 1
+				}
+			}
+		}
+	}
+
+	//Sort the preference map in order.
+	var order []int
+	for i := 0; i < candidatesNumber; i++ {
+		order = append(order, 0)
+	}
+	var index []int
+	for i := 0; i < candidatesNumber; i++ {
+		index = append(index, 0)
+	}
+	var score []int
+	for i := 0; i < candidatesNumber; i++ {
+		score = append(score, 0)
+	}
+	preferenceMap := make(map[int][]int)
+	for k, v := range votesLayoutElect {
+		j := 0
+
+		for t := 0; t < candidatesNumber; t++ {
+			order[t] = v[t]
+		}
+
+		for i := 0; i < candidatesNumber; i++ {
+
+			sum := 0
+			for i := 0; i < candidatesNumber; i++ {
```

Comment:
dont use the same iterator name in an inner for loop
The loop on line 138 also use i